pkg/kubelet/cm/node_allocatable_test.go

```diff
@@ -0,0 +1,114 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cm
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/api/v1"
+	evictionapi "k8s.io/kubernetes/pkg/kubelet/eviction/api"
+)
+
+func TestGetNodeAllocatable(t *testing.T) {
+	memoryEvictionThreshold := resource.MustParse("100Mi")
+	testCases := []struct {
+		kubeReserved   v1.ResourceList
+		systemReserved v1.ResourceList
+		capacity       v1.ResourceList
+		expected       v1.ResourceList
+		hardThreshold  evictionapi.ThresholdValue
+	}{
+		{
+			kubeReserved:   getResourceList("100m", "100Mi"),
+			systemReserved: getResourceList("50m", "50Mi"),
+			capacity:       getResourceList("10", "10Gi"),
+			expected:       getResourceList("9850m", "10090Mi"),
+		},
+		{
+			kubeReserved:   getResourceList("100m", "100Mi"),
+			systemReserved: getResourceList("50m", "50Mi"),
+			hardThreshold: evictionapi.ThresholdValue{
+				Quantity: &memoryEvictionThreshold,
+			},
+			capacity: getResourceList("10", "10Gi"),
+			expected: getResourceList("9850m", "10475274240"),
+		},
+		{
+			kubeReserved:   getResourceList("100m", "100Mi"),
+			systemReserved: getResourceList("50m", "50Mi"),
+			hardThreshold: evictionapi.ThresholdValue{
+				Percentage: 0.05,
+			},
+			capacity: getResourceList("10", "10Gi"),
+			expected: getResourceList("9850m", "10043260920"),
+		},
+
+		{
+			kubeReserved:   v1.ResourceList{},
+			systemReserved: v1.ResourceList{},
+			capacity:       getResourceList("10", "10Gi"),
+			expected:       getResourceList("10", "10Gi"),
+		},
+		{
+			kubeReserved:   getResourceList("", "100Mi"),
+			systemReserved: getResourceList("50m", "50Mi"),
+			capacity:       getResourceList("10", "10Gi"),
+			expected:       getResourceList("9950m", "10090Mi"),
+		},
+
+		{
+			kubeReserved:   getResourceList("50m", "100Mi"),
+			systemReserved: getResourceList("", "50Mi"),
+			capacity:       getResourceList("10", "10Gi"),
+			expected:       getResourceList("9950m", "10090Mi"),
+		},
+		{
+			kubeReserved:   getResourceList("", "100Mi"),
+			systemReserved: getResourceList("", "50Mi"),
+			capacity:       getResourceList("10", ""),
+			expected:       getResourceList("10", ""),
+		},
+	}
+	for idx, tc := range testCases {
+		nc := NodeConfig{
+			NodeAllocatableConfig: NodeAllocatableConfig{
+				KubeReserved:   tc.kubeReserved,
+				SystemReserved: tc.systemReserved,
+				HardEvictionThresholds: []evictionapi.Threshold{
+					{
+						Signal:   evictionapi.SignalMemoryAvailable,
+						Operator: evictionapi.OpLessThan,
+						Value:    tc.hardThreshold,
+					},
+				},
+			},
+		}
+		cm := &containerManagerImpl{
+			NodeConfig: nc,
+			capacity:   tc.capacity,
+		}
+		for k, v := range cm.GetNodeAllocatable() {
```

Comment:
this is confusing me.  see my earlier comment on the internal version of the method, but this doesn't appear correct, and at minimum is not testing both patterns where we pass true or false to the operation.